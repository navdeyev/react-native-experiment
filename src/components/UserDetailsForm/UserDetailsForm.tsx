import i18n from 'i18n-js';
import * as React from 'react';
import {Button, View} from 'react-native';
import {reduxForm, ConfigProps, Field, InjectedFormProps, FormSubmitHandler, SubmitHandler} from 'redux-form';

import {UserDetailsFormData} from '../../domains/types';
import FormTextInput from '../FormTextInput/FormTextInput';

export interface FormProps {
    handleSubmit: SubmitHandler<UserDetailsFormData, Props>;
    onSubmit: FormSubmitHandler<UserDetailsFormData>,
}

export const Form: React.SFC<FormProps> = (props) => {
    const {handleSubmit, onSubmit} = props;
    return (
        <View>
            <Field label={i18n.t('name')} name="name" component={FormTextInput}/>
            <Field label={i18n.t('username')} name="username" component={FormTextInput}/>
            <Field label={i18n.t('email')} name="email" component={FormTextInput}/>
            <Field label={i18n.t('phone')} name="phone" component={FormTextInput}/>
            <Field label={i18n.t('website')} name="website" component={FormTextInput}/>
            <Button title={i18n.t('submit')}
                    onPress={handleSubmit(onSubmit)}
                    testID="user-details-form-submit-button"
            />
        </View>
    );
};

interface Props {
    onSubmit: FormSubmitHandler<UserDetailsFormData>,
}

const UserDetailsForm: React.SFC<Props & InjectedFormProps<UserDetailsFormData, Props>> = (props) => {
    const {handleSubmit, onSubmit} = props;
    return <Form handleSubmit={handleSubmit} onSubmit={onSubmit}/>;
};

const formConfig: ConfigProps<UserDetailsFormData, Props> = {
    form: 'userDetailsForm',
};

export default reduxForm<UserDetailsFormData, Props>(formConfig)(UserDetailsForm);
