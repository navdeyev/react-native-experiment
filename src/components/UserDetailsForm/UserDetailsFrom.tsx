import i18n from 'i18n-js';
import * as React from 'react';
import {ConfigProps, Field, InjectedFormProps, reduxForm, WrappedFieldProps} from 'redux-form';
import {Button, Text, TextInput, View} from 'react-native';

interface Props extends InjectedFormProps {
}

const renderInput = (wrappedProps: WrappedFieldProps) => {
    const {onChange, value} = wrappedProps.input;
    return <TextInput onChangeText={onChange} value={value}/>;
};

const submit = (values: any) => {
    console.log('submit', values);
};

export const UserDetailsForm: React.SFC<Props> = (props) => {
    const {handleSubmit} = props;

    return (
        <View>
            <Text>{i18n.t('name')}</Text>
            <Field name="name" component={renderInput}/>
            <Field name="username" component={renderInput}/>
            <Field name="email" component={renderInput}/>
            <Field name="phone" component={renderInput}/>
            <Field name="website" component={renderInput}/>
            <Button title={i18n.t('submit')}
                    onPress={handleSubmit(submit)}
                    testID="user-details-form-submit-button"
            />
        </View>
    );
};

const formConfig: ConfigProps = {
    form: 'userDetailsForm'
};

export default reduxForm(formConfig)(UserDetailsForm);