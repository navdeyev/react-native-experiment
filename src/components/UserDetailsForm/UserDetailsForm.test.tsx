import * as React from 'react';
import {shallow} from 'enzyme';

import {Form, FormProps} from './UserDetailsForm';

describe('UserDetailsForm', () => {

    it('renders a component', () => {
        const props: FormProps = {
            onSubmit: jest.fn(),
            handleSubmit: jest.fn(() => jest.fn()),
        };
        const rendered = shallow(<Form {...props}/>);
        expect(rendered.getElement()).toMatchSnapshot();
    });

});