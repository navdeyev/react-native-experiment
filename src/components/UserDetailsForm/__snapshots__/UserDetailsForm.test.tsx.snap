// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsForm renders a component 1`] = `
<Component>
  <Field
    component={[Function]}
    label="Name"
    name="name"
  />
  <Field
    component={[Function]}
    label="Username"
    name="username"
  />
  <Field
    component={[Function]}
    label="Email"
    name="email"
  />
  <Field
    component={[Function]}
    label="Phone"
    name="phone"
  />
  <Field
    component={[Function]}
    label="Website"
    name="website"
  />
  <Button
    onPress={[MockFunction]}
    testID="user-details-form-submit-button"
    title="Submit"
  />
</Component>
`;
